core/src/main/scala/org/apache/spark/ui/UIWorkloadGenerator.scala

```diff
@@ -99,12 +103,17 @@ private[spark] object UIWorkloadGenerator {
             } catch {
               case e: Exception =>
                 println("Job Failed: " + desc)
+            } finally {
+              barrier.release()
             }
           }
         }.start
         Thread.sleep(INTER_JOB_WAIT_MS)
       }
     }
+
+    // Waiting for threads.
+    barrier.acquire()
```

Comment:
Hm yeah. Well I don't have a strong opinion either way. Either count down latch or semaphore is fine.